text/068-coqide-split.md

```diff
@@ -0,0 +1,102 @@
+---
+Title: Future of CoqIDE
+Drivers: Karl Palmskog (@palmskog), Théo Zimmermann (@Zimmi48)
+---
+
+## Summary
+
+The goal of this CEP is:
+
+1. to clarify the priority and focus shift by the Coq team away from maintaining and evolving CoqIDE;
+2. to give an opportunity to the community of CoqIDE users to assume maintenance  of and lead the future evolution of CoqIDE through a source code split out of the Coq GitHub repository.
+
+More specifically, the proposal is to make an open call for new volunteer maintainers of CoqIDE through Coq community channels, and then to proceed with a source code split only if volunteer maintainers are found.
+
+## Context and motivation
+
+[CoqIDE](https://coq.inria.fr/refman/practical-tools/coqide.html) is an Integrated Development Environment (IDE) for Coq. CoqIDE is implemented using the OCaml programming language and the GTK3 widget toolkit for Graphical User Interfaces (GUIs), thanks to the [lablgtk3](https://opam.ocaml.org/packages/lablgtk3/) OCaml package.
+
+CoqIDE communicates with Coq thanks to an [XML protocol](https://github.com/coq/coq/blob/master/dev/doc/xml-protocol.md), implemented by the coqidetop server (in the coqide-server opam package), which itself is based on a component in Coq called the [State Transaction Machine](https://coq.github.io/doc/master/api/coq-core/Stm/index.html) (STM).
+
+The drawbacks of CoqIDE and coqidetop are that:
+
+- maintaining a standalone IDE just for Coq means that a lot of standard work has to be redone and lots of standard IDE features are not implemented,
+- the XML protocol is entirely custom and suited for the CoqIDE implementation, which means that any IDE wanting to interface with Coq through this protocol has to reimplement a lot of things, and that it won't always be well-suited to the model that the IDE expects.
+
+Recently, the Coq team has changed its strategy, by focusing instead on developing support for the [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) (LSP) with minimal extensions, which should bring the possibility to develop Coq support for any standard IDE (starting with VS Code). Meanwhile, CoqIDE, coqidetop and the STM are considered outdated technology that the Coq team would like to avoid putting too much work into maintaining.
+
+Because *some users still prefer a standalone IDE for Coq*, and some are *motivated to contribute to CoqIDE*, we want to give the opportunity to the community to take over CoqIDE maintenance by splitting out the CoqIDE source code from Coq's repository, on the condition that volunteer maintainers are found.
```

Comment:
> Is this data taken from the survey?

Survey + anecdotal evidence. But you are right that we should have explicitly acknowledged the findings from the survey in this text.

Reply:
I was wondering what the users are, maybe the way to find out is to do the split.

Reply:
If we do not find volunteer maintainers, then I would really discourage doing the split as it would just make it more complex for core Coq devs to maintain CoqIDE, without bringing any additional benefits (like a new dynamics). The previous example of a split without volunteer maintainers was bignums and it was a huge mistake (until it was finally fixed by finding such volunteer maintainers years later).

Reply:
A split without maintainers would go against what we say elsewhere in the CEP:
> If no maintainers are found, CoqIDE sources will be kept in Coq's repository and minimally maintained until their eventual removal.

I think this requirement is the right thing, not least due to the experience with Bignums.